### PR TITLE
design:Implement new design for n-magnet package to use

### PR DESCRIPTION
**Pull request by Nazehs**
### Description

This Pull Request implements design updates to the newsletter sign-up button within the magnet component featured on article pages, as requested by the Content Discovery team. The magnet component alternates between displaying an event promo and offering a newsletter sign-up option. Specifically, this update focuses on enhancing the the current  design for the newsletter sign-up to align with the new design provided.

[Ticket](https://financialtimes.atlassian.net/browse/ACC-2923)

## Current Design 
image

## New Design
Screenshot 2024-02-08 at 17 31 39

## When Already Signed Up
Screenshot 2024-02-08 at 17 02 42

Screenshot 2024-02-08 at 17 06 52

## Smaller device display when newsletter has image
Screenshot 2024-02-08 at 17 07 52

## Shows upgrade button 

Screenshot 2024-02-09 at 09 56 19

## Card State demo

Screenshot 2024-02-19 at 17 06 26

[Figma Design](https://www.figma.com/file/Z63fDH0BB3NdEoO2RMyz63/Newsletter-redesign?type=design&node-id=37-265&mode=design&t=uPA5gNRqIrTT7Yzp-0)

### CODE REVIEW

File: styles/simple.scss [hunk 1]
```diff
@@ -1,18 +1,136 @@
...
+		@include oGridRespondTo($from: M) {
+			overflow: hidden;
+		}
+
+		@include oGridRespondTo($until: M) {
+			flex: 0 0 30%;
+			text-align: center;
+			color: oColorsByName('claret');
+
+		}
```
Comment (GlynnPhillips) [inline styles/simple.scss:80]: **Change request:** Here we can structure this with a mobile first responsive approach.

It would be something like this 

```suggestion
                flex: 0 0 30%;
		text-align: center;
		color: oColorsByName('claret');

		@include oGridRespondTo($from: M) {
			overflow: hidden;
			// Make any changes you need on M and bigger screens to flex, text-align and color
		}
```

File: styles/simple.scss [hunk 2]
```diff
@@ -1,18 +1,136 @@
...
+	}
+
+	.n-newsletter-signup__heading {
+		@include oTypographyDisplay($scale: 2, $weight: 'bold');
+
+		display: inline-block;
+		align-self: flex-end;
+		margin: 0;
+
+		// for IE<11
```
Comment (GlynnPhillips) [inline styles/simple.scss:90]: **Question:** Are these needed? Do we still support IE11?

File: styles/simple.scss [hunk 3]
```diff
@@ -1,18 +1,136 @@
...
+
+		@include oGridRespondTo($until: S) {
+			font-size: 40px;
+			line-height: 40px;
+		}
+
+		@include oGridRespondTo($from: S, $until: M) {
+			font-size: 20px;
+			line-height: normal;
+		}
```
Comment (GlynnPhillips) [inline styles/simple.scss:109]: **Change request:** Please structure this with a mobile first approach so you are only overriding styles at each break point. This normally means you don't need to do stuff like From S until M you would just do override at the M point

File: styles/simple.scss [hunk 4]
```diff
@@ -1,18 +1,136 @@
...
-		font-weight: 600;
-		vertical-align: middle;
+		@include oTypographySerif($scale: 1);
+		display: block;
+		flex-grow: 1;
+		font-family: Metric;
+		font-size: 14px;
+		font-style: normal;
+		font-weight: 400;
+		line-height: 16px;
```
Comment (GlynnPhillips) [inline styles/simple.scss:121]: **Change request:** You are using `@include oTypographySerif($scale: 1);` above but then changing every aspect of what that provides. I think we can stick to using the origami helpers here for these styles

File: styles/simple.scss [hunk 5]
```diff
@@ -1,18 +1,136 @@
...
+		color: oColorsByName("black-80");
+
+		@include oGridRespondTo($from: S, $until: M) {
+			margin-top: 0;
+		}
```
Comment (GlynnPhillips) [inline styles/simple.scss:126]: **Change Request:** Here we can structure this with mobile first approach. Something like the below (this is untested)

```sass
margin-top: 0;
@include oGridRespondTo($from: M) {
    margin-top: // whatever it should be on M+ screens
}


```

File: templates/partials/newsletter-form.html [hunk 6]
```diff
@@ -1,51 +1,61 @@
...
+  "
+  method="POST"
+>
+  {{> n-myft-ui/components/csrf-token/input}}
+<div
+  class="ft-concept-button ft-concept-button--follow ft-concept-button--standard"
+  data-ft-concept-button-aria-live-pressed-text="Now following {{name}} on my FT"
+  data-ft-concept-button-aria-live-unpressed-text="No longer following {{name}} on my FT"
+  data-ft-concept-button-aria-label-pressed-text="Stop following {{name}} on my FT"
+  data-ft-concept-button-aria-label-unpressed-text="Follow {{name}} on my FT"
```
Comment (GlynnPhillips) [inline templates/partials/newsletter-form.html:20]: **Question / Change Request:** It looks like these are unrelated to newsletters and are for myFT follows, is there a reason they are there? If no we can remove / update them to be correct for newsletters

File: templates/partials/newsletter-form.html [hunk 7]
```diff
@@ -1,51 +1,61 @@
...
+          Unsubscribe from {{name}}
+        {{~else~}}
+          Sign Up subscribe to {{name}}
+        {{~/if~}}
+      "
+      title="
+        {{~#if userIsSubscribed~}}
+          Unsubscribe from {{name}}
+        {{~else~}}
+          Subscribe to {{name}}
```
Comment (GlynnPhillips) [inline templates/partials/newsletter-form.html:38]: **Comment:** Some places we are using "sign up" and here we are using "subscribe". Please can we check which is the correct wording and then up date accordingly to be consistent

File: templates/partials/newsletter-form.html [hunk 8]
```diff
@@ -1,51 +1,61 @@
...
+      "
+      data-trackable="
+        {{~#if userIsSubscribed~}}
+          newsletter-unsubscribe
+        {{~else~}}
+          newsletter-subscribe
+        {{~/if~}}
+      "
+      type="submit"
+      aria-describedby="feedback-message__newsletter-{{id}}"
```
Comment (GlynnPhillips) [inline templates/partials/newsletter-form.html:49]: **Question:** I can't see where the element with the id `feedback-message__newsletter-{{id}}` is created, do you know?

File: templates/simple.html [hunk 9]
```diff
@@ -1,9 +1,48 @@
...
+    <div class="newsletter__flex-child-left">
+        <picture>
+            <source srcSet="{{imageUrl}}" />
+            <img
+                src="{{imageUrl}}"
+                alt=""
+                data-component="newsletter-picture"
+            />
+        </picture>
+    </div>
```
Comment (GlynnPhillips) [inline templates/simple.html:17]: **Change request:** Here we are missing some indentation to put it inside the `if`

File: templates/simple.html [hunk 10]
```diff
@@ -1,9 +1,48 @@
...
+            <h2 class="n-newsletter-signup__heading">
+                {{name}}
+                <span class="o-normalise-visually-hidden">
+                    {{#if isPremium}}, premium{{/if}}
+                    {{#if frequency}}, {{frequency}}{{/if}}
+                </span>
+            </h2>
+            
+            {{#if isPremium}}
+                <span aria-hidden="true" class="n-newsletter-signup__top-meta o-labels o-labels--content-premium">Premium</span>
```
Comment (GlynnPhillips) [inline templates/simple.html:30]: **Question:** Why is this marked as hidden via `aria-hidden="true"`? It appears to have styles applied to it that doesn't suggest its hidden

**Review by GlynnPhillips — CHANGES_REQUESTED** [review]

**Comment by GlynnPhillips** [comment]
@Nazehs after reviewing I had a realisation that `n-newsletter-signup` is referenced in your code a lot and thats also an existing [component](https://github.com/Financial-Times/n-newsletter-signup/blob/main/styles/newsletter.scss).

A few questions go through my head about this

1. Is much of this copy and pasted from that component?
2. is there a reason why we aren't doing this work by including that component?
3. If we are not using that component we should not mimick its classnames because thats confusing to debug in the future.

Let me know if you want to talk about this

**Comment by Nazehs** [comment]
> @Nazehs after reviewing I had a realisation that `n-newsletter-signup` is referenced in your code a lot and thats also an existing [component](https://github.com/Financial-Times/n-newsletter-signup/blob/main/styles/newsletter.scss).
> 
> A few questions go through my head about this
> 
> 1. Is much of this copy and pasted from that component?
> 2. is there a reason why we aren't doing this work by including that component?
> 3. If we are not using that component we should not mimick its classnames because thats confusing to debug in the future.
> 
> Let me know if you want to talk about this
@GlynnPhillips 
Let's discuss this further if my interpretation of your question isn't accurate. The task for this ticket involved updating the existing newsletter designs to a new style. Included in the package are two versions of the newsletters: the standard newsletter and the simple newsletter design. The latter required more significant changes and my work was around the latter primarily.

File: templates/partials/newsletter-form.html (hunk 8 above)
Comment (Nazehs) [reply templates/partials/newsletter-form.html:49]: The "aria-describedby" attribute, identified by the given ID, is utilized to provide feedback once the processing of the newsletter request, which could result in either success or error, is complete.

File: templates/partials/newsletter-form.html (hunk 6 above)
Comment (Nazehs) [reply templates/partials/newsletter-form.html:20]: I am utilizing the "Follow FT" concept button, and my understanding is that by signing up, users will also be able to view it in their "My FT" section, similar to current newsletters. This effectively acts as a newsletter sign-up. I am open to making adjustments if you suggest a better alternative.

File: styles/simple.scss [hunk 11]
```diff
@@ -1,18 +1,136 @@
...
+
+		@include oGridRespondTo($from: M) {
+			flex-direction: row;
+			align-items: center;
+
+		}
+
+		h1 {
+			padding-left: 0;
+		}
```
[the remark this answers is not in the document]
Comment (Nazehs) [reply styles/simple.scss:38]: Good catch. The style mentioned above shouldn't be there since I switched to using an `<h2>` tag in the newsletter section. I'll remove it from the styles.

File: styles/simple.scss [hunk 12]
```diff
@@ -1,18 +1,136 @@
...
+	flex-shrink: 0;
 	border: 0;
+	gap: 20px;
+	padding-top: 20px;
+	background-color: oColorsByName("white-80");
+
+	@include oGridRespondTo($from: S, $until: M) {
+		padding-left: 24px;
+		padding-bottom: 0;
+	}
```
[the remark this answers is not in the document]
Comment (Nazehs) [reply styles/simple.scss:13]: Noted. I will make the adjustments

File: templates/simple.html (hunk 9 above)
Comment (Nazehs) [reply templates/simple.html:17]: Noted. I will make the change

File: templates/simple.html (hunk 10 above)
Comment (Nazehs) [reply templates/simple.html:30]: My understanding of `aria-hidden="true"` is that it's used primarily for aesthetic purposes, to hide elements from assistive technologies that don't need to be read aloud or interacted with, which is why it's set to true. I didn't set this personally because it was there on the existing design which i still maintained it.

File: templates/partials/newsletter-form.html (hunk 6 above)
Comment (GlynnPhillips) [reply templates/partials/newsletter-form.html:20]: Unfortunately following something in myFT is not the same as signing up to a newsletter so this will need to be changed to reflect that this is a newsletter sign up

File: templates/partials/newsletter-form.html (hunk 8 above)
Comment (GlynnPhillips) [reply templates/partials/newsletter-form.html:49]: Sorry maybe my question wasn't as good as it could be. The `aria-describedby` is pointing to an element with an `id` attribute that equals `feedback-message__newsletter-{{id}}` where is that element created?

File: templates/simple.html (hunk 10 above)
Comment (GlynnPhillips) [reply templates/simple.html:30]: Your understanding is correct but if its visible to sighted users then its likely it should be present for users who are using assistive tech. Why would a user who can't visually see the page not need to know this is a Premium newsletter? Do they get that information from somewhere else?

**Comment by GlynnPhillips** [comment]
Sorry @Nazehs ignore that last comment for some reason I thought we weren't in n-newsletter-signup 🤦 I think because there was lots of references in conversations to n-magnet and next-article I got confused 👎

File: templates/partials/newsletter-form.html (hunk 6 above)
Comment (GlynnPhillips) [reply templates/partials/newsletter-form.html:20]: Also a follow up question, what is using these `data` attributes?

File: templates/partials/newsletter-form.html [hunk 13]
```diff
@@ -1,51 +1,61 @@
...
+      {{#if updatingPreference}}disabled="disabled" aria-disabled="true"{{/if}}
+    >
+      {{#if updatingPreference}}
+        Updating<span class="o-normalise-visually-hidden">&nbsp;{{name}} subscription</span>…
```
Comment (GlynnPhillips) [inline templates/partials/newsletter-form.html:53]: **Question:** I wonder if the use of "subscription" is confusing here as this might make users think that we are changing their subscription.

Maybe something like the following would be clearer? `Updating<span class="o-normalise-visually-hidden">&nbsp;{{name}} newsletter preference`

File: styles/simple.scss [hunk 14]
```diff
@@ -1,18 +1,136 @@
...
+	.n-newsletter-signup__top-meta {
+		top: 0;
+		right: 0;
+	}
+
+
+
+	img {
+		max-width: 100%;
+	}
```
[the remark this answers is not in the document]
Comment (Nazehs) [reply styles/simple.scss:64]: I will change this but the tag is the only image in the newsletter and its specific to `.n-newsletter-signup--simple`

File: templates/partials/newsletter-form.html (hunk 8 above)
Comment (Nazehs) [reply templates/partials/newsletter-form.html:49]: This element is created here https://github.com/Financial-Times/n-newsletter-signup/blob/main/js/feedback-messaging.js#L6 which is used for the feedback

File: templates/simple.html (hunk 10 above)
Comment (Nazehs) [reply templates/simple.html:30]: This i am not aware but i don't think they are getting it somewhere. I will remove it.

File: templates/partials/newsletter-form.html (hunk 8 above)
Comment (GlynnPhillips) [reply templates/partials/newsletter-form.html:49]: Oh thanks Nazeh. For some reason that wasn't returned in my search :)

File: templates/partials/newsletter-form.html (hunk 6 above)
Comment (Nazehs) [reply templates/partials/newsletter-form.html:20]: My bad With the data attribute I Kind of confused myself with my ft :D looking at the code now.

Regarding the data attribute, the data serves an accessibility purpose, as recommended by Origami team, to ensure users are informed about the actions that have taken place.

https://github.com/Financial-Times/origami/tree/ft-concept-button-v1.2.2/components/ft-concept-button

File: templates/simple.html (hunk 10 above)
Comment (GlynnPhillips) [reply templates/simple.html:30]: Looks like the reason its marked as `aria-hidden` is that the data is also in the headline so doesn't need to be duplicated.

Personally this looks a bit of a messy scenario but you can probably continue without changing this for now

Screenshot 2024-02-12 at 14 05 48

Comment (Nazehs) [reply templates/simple.html]: That makes sense to me now, too. However, since the current design does not capture any such data, it would be reasonable to remove it to ensure screen readers can identify it as premium content.

Comment (GlynnPhillips) [reply templates/simple.html]: > However, since the current design does not capture any such data, it would be reasonable to remove it to ensure screen readers can identify it as premium content.

Can you explain this to me please I am not sure I understand what you mean here?

File: styles/newsletter.scss [hunk 15]
```diff
@@ -12,92 +10,98 @@ $o-grid-gutter-small: oGridGutter(S);
...
-		line-height: 28px;
+
+	.n-newsletter-signup__header {
+		display: flex;
+		position: relative;
+		box-sizing: border-box;
+		min-height: 100px;
+		padding: 5px 12px;
+		background-color: oColorsByName('claret');
+		color: white;
```
Comment (GlynnPhillips) [inline styles/newsletter.scss:24]: **Change request:** Here can we use `oColorsByName` or maybe a hex code `#ffffff` instead of `white`

Comment (GlynnPhillips) [reply styles/newsletter.scss:24]: @Nazehs This is still waiting to be actioned

File: styles/newsletter.scss [hunk 16]
```diff
@@ -12,92 +10,98 @@ $o-grid-gutter-small: oGridGutter(S);
...
+
+		@include oGridRespondTo($from: M, $until: XL) {
+			font-size: 24px;
+			line-height: 28px;
+		}
+
+		@include oGridRespondTo($until: S) {
+			font-size: 40px;
+			line-height: 40px;
+		}
```
Comment (GlynnPhillips) [inline styles/newsletter.scss:50]: **Comment:** These have not been formatted to be mobile first as suggested in a previous reviews. Please reach out to the team if you need help with this

File: styles/newsletter.scss [hunk 17]
```diff
@@ -12,92 +10,98 @@ $o-grid-gutter-small: oGridGutter(S);
...
+	.n-newsletter-signup--subscribed .n-newsletter-signup__submit {
+		@include oButtonsContent($opts: ('type': 'primary',
+				'icon': 'tick',
+			), $include-base-styles: false);
+		margin: 16px 16px 16px 0;
+	}
+
+	.n-newsletter-signup__upgrade {
+		@include oButtonsContent($opts: ('type': 'secondary',
+			));
```
Comment (GlynnPhillips) [inline styles/newsletter.scss:102]: **Comment:** Looks like a formatting issue here maybe?

File: styles/newsletter.scss [hunk 18]
```diff
@@ -12,92 +10,98 @@ $o-grid-gutter-small: oGridGutter(S);
...
-}
-
-.n-newsletter-signup__upgrade {
-	@include oButtonsContent($opts: (
-		'type': 'secondary',
-		'size': 'big'
-	));
-	margin: 16px 16px 16px 0;
-}
+	.n-newsletter-signup--subscribed .n-newsletter-signup__submit {
```
Comment (GlynnPhillips) [inline styles/newsletter.scss:93]: **Change request:** With BEM there should be much need for nesting but here we are also nesting in 2 different styles. Once by nesting within the parent styles

```css

.parent {
    .nested: {
    }
}
```

and then once by chaining the classname selectors

```css
.parent .nested {}
```
Lets please check if we need to nest at all or whether BEM can prevent this from being needed and then also have a consistency to our nesting style if it is needed.

File: styles/simple.scss [hunk 19]
```diff
@@ -1,18 +1,106 @@
...
+
+
+	.n-newsletter-signup__flex-child-left {
+		position: relative;
+		text-align: center;
+		color: oColorsByName('claret');
+		flex: 0 0 100%;
+		max-width: 158px;
+		max-height: 89px;
```
Comment (GlynnPhillips) [inline styles/simple.scss:54]: **Question:** These feel like very specific numbers, do you know why they are so specific?

File: styles/simple.scss [hunk 20]
```diff
@@ -1,18 +1,106 @@
...
+
+		@include oGridRespondTo($from: M) {
+			max-width: 158px;
+			max-height: 89px;
```
Comment (GlynnPhillips) [inline styles/simple.scss:58]: **Change request:** These are repeats of the default styles above so they are not needed

File: styles/simple.scss [hunk 21]
```diff
@@ -1,18 +1,106 @@
...
+		}
+	}
+
+	.n-newsletter-signup__top-meta {
+		top: 0;
+		right: 0;
+	}
+
+
+	.n-newsletter-signup__flex-child-left {
```
Comment (GlynnPhillips) [inline styles/simple.scss:48]: **Question:** `flex-child-left` describes the styles/layout rather than what that element is. Is there a reason for this naming? does that element contain different things at different times?

Comment (GlynnPhillips) [reply styles/simple.scss:48]: **Question:** I also cannot see these styles being used anywhere in the demos so can you explain where they are used. If they are used we need a demo for them and if they are not then they can be removed

File: styles/simple.scss [hunk 22]
```diff
@@ -1,18 +1,106 @@
...
+		@include oGridRespondTo($from: M) {
+			max-width: 158px;
+			max-height: 89px;
+			flex: 0 0 30%;
+			overflow: hidden;
+		}
+
+		&__card_image {
+			max-width: 100%;
+		}
```
Comment (GlynnPhillips) [inline styles/simple.scss:65]: Question: I also cannot see these styles being used anywhere in the demos so can you explain where they are used. If they are used we need a demo for them and if they are not then they can be removed


**Change Request:** Using `&` here is not consistent. It also introduces a couple of issues the first one being it will produce a style name that is not BEM as the style name will be `.n-newsletter-signup__flex-child-left__card_image`

Also I would encourage us to not use `&` because it makes the code harder to read and harder to search for styles.

File: styles/simple.scss [hunk 23]
```diff
@@ -1,18 +1,106 @@
...
+	}
+
+	.n-newsletter-signup__heading {
+		@include oTypographyDisplay($scale: 2, $weight: 'bold');
+		
+		display: inline-block;
+		align-self: flex-end;
+		margin: 0;
+		font-size: 20px;
+		line-height: normal;
```
Comment (GlynnPhillips) [inline styles/simple.scss:75]: **Change request:** Here we are overriding the styles set by `oTypographyDisplay` which we shouldn't be doing so we need to remove these styles and just use the ability to set font size via `oTypographyDisplay `

File: styles/simple.scss (hunk 4 above)
Comment (GlynnPhillips) [reply styles/simple.scss:91]: @Nazehs you have not actioned this change request

File: styles/simple.scss (hunk 5 above)
Comment (GlynnPhillips) [reply styles/simple.scss:96]: @Nazehs this still needs actioning

File: templates/partials/newsletter-form.html [hunk 24]
```diff
@@ -1,51 +1,61 @@
...
+  "
+  method="POST"
+>
+  {{> n-myft-ui/components/csrf-token/input}}
+<div
+  class="ft-concept-button ft-concept-button--follow ft-concept-button--standard"
+  data-ft-concept-button-aria-live-pressed-text="Now following {{name}} newsletter"
+  data-ft-concept-button-aria-live-unpressed-text="No longer following {{name}} newsletter"
+  data-ft-concept-button-aria-label-pressed-text="Stop following {{name}} newsletter"
+  data-ft-concept-button-aria-label-unpressed-text="Follow {{name}} newsletter"
```
Comment (GlynnPhillips) [inline templates/partials/newsletter-form.html:20]: **Change request:** The wording on this about "following" a newsletter is still incorrect as we don't "follow" newsletters. This language is inconsistent with the language used elsewhere in this feature regarding "Sign up", "Subscribe" etc

File: styles/simple.scss (hunk 21 above)
Comment (GlynnPhillips) [reply styles/simple.scss:48]: Ok I see it being used [here](https://github.com/Financial-Times/n-newsletter-signup/pull/107/files#diff-71e84ed882ed5c3832fcc269b1e324b48c71721cf9de414e8a4e6ae72e9f441dR8) so can we please add a demo for this

File: templates/simple.html (hunk 10 above)
Comment (Nazehs) [reply templates/simple.html]: In the previous design, the word `premium` appeared as text within the newsletter's content, but it is no longer present in the current design. This is why I mentioned earlier that it makes sense to remove the aria label now. Initially, the removal was to prevent screen readers from announcing it twice, but this concern no longer applies.

Comment (GlynnPhillips) [reply templates/simple.html]: Sorry I don't think my previous response with the screenshot was clear. In your new design the word premium is still included twice. Its in the title its just visually hidden. So it does still make sense to `aria-hidden` one of those 2 instances

Comment (Nazehs) [reply templates/simple.html:31]: I see what you mean.

**Review by GlynnPhillips — CHANGES_REQUESTED** [review]
@Nazehs I believe some of the demos are still missing on this, can we ensure that all different layouts have a demo please

**Comment by Nazehs** [comment]
> @Nazehs I believe some of the demos are still missing on this, can we ensure that all different layouts have a demo please

Apologies, @GlynnPhillips, could you specify which demo you're referring to? I believe all demos for the two card designs are accounted for. There are only two states/designs for the new card: one with an image and one without, as depicted in the screenshot. For the image to be displayed, the `imageUrl` must be included in the newsletter data. Please let me know if you are referring to something else.

File: styles/newsletter.scss [hunk 25]
```diff
@@ -30,23 +30,19 @@ $o-grid-gutter-small: oGridGutter(S);
...
 			font-size: 24px;
 			line-height: 28px;
 		}
 
-		@include oGridRespondTo($until: S) {
-			font-size: 40px;
-			line-height: 40px;
+		@include oGridRespondTo($from: M) {
+			font-size: 30px;
+			line-height: 30px;
```
[the remark this answers is not in the document]
Comment (Nazehs) [reply styles/newsletter.scss:45]: Regarding the discussion on the change request with designer, no design modifications were made primarily because they were not included in the scope of this ticket. Consequently, there isn't a specific designer available to consult on this matter, unless we consider reaching out to whoever was responsible for the original design years ago. Nevertheless, I am okay to implementing adjustments that align as closely as possible with the current standards where feasible.

**Comment by GlynnPhillips** [comment]
@Nazehs we should have demos for all of the scenarios you have just mentioned such as the images existing or not existing. An example is I don't see any demo for when the image appears. If there are any conditions that should result in an expected outcome then that should be documented in the demos

Screenshot 2024-02-19 at 09 59 54

**Comment by Nazehs** [comment]
> @Nazehs we should have demos for all of the scenarios you have just mentioned such as the images existing or not existing. An example is I don't see any demo for when the image appears. If there are any conditions that should result in an expected outcome then that should be documented in the demos
> 
> Screenshot 2024-02-19 at 09 59 54

@GlynnPhillips, I've updated the newsletter parameters in the readme to incorporate 'imageUrl' as an optional data point, which is required for the functionality mentioned above. Given this, I believe we don't need to prepare a demo to cover that states. However, creating a demo would necessitate significant modifications to the handlebars, and recalling our previous discussions about transitioning to React, proceeding with these changes might not be the best use of our efforts.

**Comment by GlynnPhillips** [comment]
@Nazehs I am afraid I disagree that the demos are not needed. Demos are there to help visualise all scenarios whether they are optional or not.

Having those demos from the start would have helped me to review this PR and they will help others in the future whether its adding new features or debugging.

I also don't see how adding a couple of small demos would result in significant modifications to handlebars

File: demos/app.js [hunk 26]
```diff
@@ -49,10 +49,12 @@ app.engine('.html', new PageKitHandlebars({
 app.use('/public', express.static(path.join(__dirname, '../public'), { redirect: false }));
 
 app.get('/', (req, res) => {
+	const state = req.query.state || 'default';
+	const cardState = fixtures[state];
```
Comment (GlynnPhillips) [inline demos/app.js:53]: **Suggestion:** Sorry to be a pain but I think this might be over complicating the demos. I think we can just pass all data into the template and have a single page that renders all variations. 

I will try and knock up a quick branch to explain what I mean

File: demos/demo.html [hunk 27]
```diff
@@ -1,29 +1,33 @@
...
+	<title>demo</title>
+	<link rel="stylesheet" href="/public/main.css">
+</head>
+
+<body>
+	<div class="o-grid-container o-grid-container--snappy">
+		<div class="o-grid-row">
+			<h1>n-newsletter-signup demo</h1>
+		</div>
+		<div class="o-grid-row">
```
Comment (GlynnPhillips) [inline demos/demo.html:14]: **Change request:** This results in nested `o-grid-rows` which I don't believe you can do. The nested one is on [line 23](https://github.com/Financial-Times/n-newsletter-signup/pull/107/commits/e10cbce67c077ec62d4532ab51b4bbafb6f43bfe#diff-7ced4bcd2a7e995789ee3c6d8459095a96ed232b6944f01702dac58842f16335R23)

It also results in the `h2` with the `Simple inline` text not rendering in the correct place

Screenshot 2024-02-20 at 10 27 34

Comment (Nazehs) [reply demos/demo.html:14]: I have fix the missing closing div tag.

**Review by GlynnPhillips — APPROVED** [review]
Thanks for your patience on all the changes :)
